### DISCUSSION TITLE
Update record to something different than null using relation queries

### DISCUSSION THREAD

**Discussion by ScreamZ** [prisma/prisma#24887] [Q&A]
### Question

Hello,

I have the following code: 

```prisma
model Player {
  id        String   @id @default(cuid())
  name      String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  post     Post
}

model Post {
  id        String   @id @default(cuid())
  content   String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  player    Player   @relation(fields: [playerId], references: [id])
  playerId  String
}
```
When deleting a player, I would like to replace post playerID by the first other user different than the current I'm deleting. (When i say first, I could also specify an ID)

I'm trying to play with https://www.prisma.io/docs/orm/prisma-client/queries/relation-queries#update-all-related-records-or-filter but can't succeed.

```ts
 where: { id: toDeleteID },
        data: {
          post: { connect: { playerId: { notIn: [toDeleteID] } } },
        },
````
gives  is not assignable to type `undefined` in Typescript

Maybe anyone could help me ?

Thanks

**Comment by ludralph** [comment]
HI @ScreamZ 👋 

Thank you for raising this question. 

The schema you shared has an error in it
<img width="822" alt="image" src="https://github.com/user-attachments/assets/96ba159a-747c-4c59-9515-9519303a6642">
```
Error parsing attribute "@relation": The relation field `player` on Model `Post` is required. This is not valid because it's not possible to enforce this constraint on the database level. Please change the field type from `Player` to `Player?` to fix this.
```
Is the relationship between `Post` and `Player` a one-to-one or one-to-many? Understanding the correct relationship will help us evaluate this further.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
